build/index.js

```diff
@@ -228,13 +228,29 @@ async function buildDocumentFromURL(url) {
 async function buildLiveSamplePageFromURL(url) {
   const [documentURL, sampleID] = url.split("/_samples_/");
   const document = Document.findByURL(documentURL);
-  const liveSamplePage = kumascript.buildLiveSamplePage(
-    document.url,
-    document.metadata.title,
-    (await kumascript.render(document.url))[0],
-    { id: sampleID, createFlaw() {} }
-  );
-  return liveSamplePage.html;
+  if (!document) {
+    return `unable to find ${documentURL}`;
+  }
+  // Convert the lower-case sampleID we extract from the incoming URL into
+  // the actual sampleID object with the properly-cased live-sample ID.
+  for (const sampleIDObject of kumascript.getLiveSampleIDs(
+    document.metadata.slug,
+    document.rawHtml
+  )) {
+    if (sampleIDObject.id.toLowerCase() === sampleID) {
+      const liveSamplePage = kumascript.buildLiveSamplePage(
+        document.url,
+        document.metadata.title,
+        (await kumascript.render(document.url))[0],
+        sampleIDObject
+      );
+      if (liveSamplePage.flaw) {
+        return liveSamplePage.flaw.errorMessage;
```

Comment:
also something we should `throw` instead of `return`ing it (and, optimally also handle in server)